apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/amqp_data_consumer.ex

```diff
@@ -164,57 +192,60 @@ defmodule Astarte.DataUpdaterPlant.AMQPDataConsumer do
 
       :invalid_msg ->
         # ACK invalid msg to discard them
-        Basic.ack(chan, meta.delivery_tag)
+        @adapter.ack(chan, meta.delivery_tag)
     end
 
-    {:noreply, chan}
+    {:noreply, state}
   end
 
-  def handle_info({:async_init, queue_name}, _state) do
-    # This will block until Rabbit connection is up
-    rabbitmq_connection = ConnectionManager.get_connection()
+  defp schedule_connect() do
+    Process.send_after(self(), :connect, @reconnect_interval)
+  end
 
-    {:ok, chan} = initialize_chan(rabbitmq_connection, queue_name)
+  defp do_connect(state) do
```

Comment:
I'd change the naming given you don't actually connect anymore here, maybe `init_consume` or something like that